### ISSUE TITLE
OFI BTL appears to try to open a TX context with no capabilities, then crashes.

### ISSUE THREAD

**Issue by mwheinz** [open-mpi/ompi#9200]
Thank you for taking the time to submit an issue!

## Background information

We're doing final debugging and testing of a new OFI provider and we ran into this issue with Open MPI 4.1.x. I'm guessing it exists in other branches as well. It's possible the problem will turn out to be the new provider, not OMPI, I'm just beginning to investigate.

OMPI was built from a git clone:

```
./configure --with-hwloc=internal --with-libevent=internal --with-pmix=internal --with-ofi --with-ucx=no 
--with-usnic=no --with-ugni=no --with-psm2=yes --with-verbs=no --with-psm=no 
--enable-debug --prefix=/usr/mpi/gcc/openmpi-v4.1.x
```

Running a simple test program from the MPICH test suite, we hit the following output. The lines beginning "MHEINZ" are print statements I added to either OMPI or OFI to try and trace the problem:

```
libfabric:1700424:opx:fabric:fi_opx_ref_init():212<debug> initializing ref count for (scalable endpoint) to (0)
MHEINZ: caps = 0x0000000000000000
libfabric:1700424:opx:domain:fi_opx_tx_ctx():151<debug> TX capabilities not specified for TX context
MHEINZ: context = 0x1af9dc0, 1
MHEINZ: context->cq = (nil)
MHEINZ: context->rdma_comp_list = 0x42
attrt: btl_ofi_context.c:338: mca_btl_ofi_context_finalize: Assertion `OPAL_OBJ_MAGIC_ID == ((opal_object_t *) (&context->rdma_comp_list))->obj_magic_id' failed.
[cn-priv-01:1700424] *** Process received signal ***
[cn-priv-01:1700424] Signal: Aborted (6)
[cn-priv-01:1700424] Signal code:  (-6)
```

So, basically, it looks like the OFI BTL is using an empty capabilities list, which the OFI provider rejects, which then causes the BTL to try and free the rdma_comp_list which had never actually been allocated. (I assume 0x42 is just the uninitialized contents of the pointer.)

Per our recent conversations on "who is the maintainer of the OFI BTL and MTL" I'm going to continue looking into this.

**Comment by mwheinz** [comment]
Looks like PSM2 didn't care what capabilities were requested, it just provided a tx_context. Probably why this was never a problem before.

**Comment by mwheinz** [comment]
Okay - at this point I'm pretty sure the problem is in the provider not in Open MPI. I'll re-open if it turns out there's work to be done in the BTL.

**Comment by rajachan** [comment]
FWIW, you can tag @open-mpi/ofi to get the attention of folks who, as Howard/Jeff put it, collectively maintain the OFI MTL and BTL.
